docs/3-cloud-computing/3.2.5-lambda.md

```diff
@@ -0,0 +1,125 @@
+# AWS Lambdas
+
+AWS Lambda is a serverless compute service provided by Amazon Web Services (AWS). It allows developers to run their code without provisioning or managing servers, and only pay for the compute time consumed. Lambdas are event-driven, meaning they are triggered by specific events, such as a user uploading a file to S3 or a change in a DynamoDB table. They can also be used to build and run microservices, perform real-time data processing and analytics, and handle backend tasks such as authentication and authorization. Additionally, Lambdas can be integrated with other AWS services, such as API Gateway, S3, and Kinesis, making it a versatile and powerful tool for building scalable, fault-tolerant applications.
+
+## Exercise - Creating a Lambda function with a trigger
+
+### Overview
+- In this exercise we will be using s3 buckets to upload a csv file to a lambda function, and update a DynamoDB table.
+
+### Steps
+
+1. Set up an s3 bucket.
+
+2. Set up a DynamoDb table
+
+3. Write a python program to grab items from a csv and push to a DynamoDB table
+    - Make sure to have the right permissions
+    - Add a trigger of type s3 bucket
+        - Specify an event type
+            - Hint: Think of what kind of files you will be uploading to the bucket
+
+    - Make a lambda function that grabs every item from a csv file
+
+    Here is some helpful documentation for [boto3](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/dynamodb.html)
+
+``` python
+
+import json
+import boto3
+import csv
+
+      # lambda function handler here
+
+   record_list = []
+   
+   try:
+      # boto3 resources for s3 here
+
+      # boto3 resource for dynamodb here
+       
+      table = dynamodb.Table('** DYNAMO-DB-TABLE-NAME ***')
+      bucket = event['Records'][0]['s3']['bucket']['name']
+      key = event['Records'][0]['s3']['object']['key']
+      s3_object = s3_resource.Object(bucket,key)
+      
+      data = s3_object.get()['Body'].read().decode("utf-8").splitlines()
+      lines=csv.reader(data)
+      headers=next(lines)
+      
+      for user_data in lines:
+         print(user_data)
+
+         # Create table items here
+      
+   except Exception as e:
+      print(str(e))
+            
+   return {
+      'statusCode': 200,
+      'body': json.dumps('CSV success')
+   }
+
+```
+
+4. Once your python code is ready, download this csv file [zillow.csv](img3/zillow.csv)
```

Comment:
This link doesn't work for me locally, I end up with a 404. I'm not sure why that is, could just be a local issue, but it's probably worth checking on.